source/develop/release-management/features/virt/enhance-import-export-with-ova.md

```diff
@@ -180,81 +170,71 @@ Note:
+3. Backend asks VDSM to create an OVA file with the complete OVF 
+   specification and export destination.
+4. VDSM runs ovirt-ova tool - creates the OVA in the destination
+   (using qemu-img convert which may include file format conversion).
+   
 Notes:
 
-1. Step 5 is equivalent to `qemu-img convert` with the collapse option
+1. Step 4 is equivalent to `qemu-img convert` with the collapse option
    set and the target format set to QCOW.
-2. The OVF configuration needs to be streamed to the beginning of the
-   OVA file (according to the OVA specification, the OVF should be the
-   first tar entry inside the OVA file).
-3. The back end should be provided with the overall amount of data
-   written to the disk entry in step 4, because this information needs
-   to be set inside the OVF configuration.
-4. The OVA file will be created with a special extension (step 3). At
-   the end of the process (step 6), the file will be renamed, and the
-   extension dropped. That is done in order to enable tools that
-   synchronize shares between various deployments to differentiate
-   partial and complete files.
+2. Only after building the OVA file we could update the ovf with the
+   export size.
+3. If possible, we'll consider using streaming the disk data directly
+   from the qemu chain into OVA.
+4. In future openshift env we could use the ovirt-ova tool to run it
+   as an openshift job witout depending on VDSM, or using ansible.
 
 ## An Enhancement for Import OVAs
 Importing an OVA that was generated by oVirt won't involve virt-v2v and
 will be similar to the upload-ova process:
 
 ![](../../../../images/wiki/import-ova.png)
 
-1. The front end sends the back end a VM and a path on a host.
-2. The back end retrieves the OVF configuration from the OVA using an
-   agent located on the host (Imageio/Other).
-3. The back end parses the OVF configuration and returns a VM to the
-   front end.
-4. The front end sends (optionally customized) parameters of the import
-   process to the back end.
-5. The back end persists the VM into the database.
-6. The back end triggers the creation of the target VM disks on the
+1. The frontend sends the backend a VM and a path on a host.
+2. The backend retrieves the OVF configuration from the OVA using vdsm.
+   vdsm will run the ova-ovirt tool.
+3. The backend parses the OVF configuration and returns a VM to the
+   frontend.
+4. The frontend sends (optionally customized) parameters of the import
+   process to the backend.
+5. The backend persists the VM into the database.
+6. The backend triggers the creation of the target VM disks on the
    storage domain.
-7. The back end sends the import parameters, including the destination
-   disks, to imageio-proxy, that sends them to imageio.
-8. Imageio reads the disk from the OVA and according to the import
-   parameters streams the data into the corresponding target disks.
+7. The backend sends the import parameters, including the destination
+   disks to vdsm.
```

Comment:
I don't see any reason to use ansible for this flow. Uploading/downloading images require two components: vdsm for control, imageio for doing the io. The suggested vdsm change is very minimal, I don't see any problem with it.

I think our goal is to create a reusable component, that we can run later on openshift as a job without engine or vdsm. Sending a request to vdsm and starting the tool from vdsm is the minimal change to make this work with the current system, the tool itself should be able to work without change given right options.